File: CS/LegendItemCreatingExample/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.Utils;
using DevExpress.XtraMap;

namespace LegendItemCreatingExample {
    public partial class Form1 : Form {
        const string filePath = @"../../Countries.shp";
        
        public Form1() {
            InitializeComponent();

            InitializeMap();
        }

        private void InitializeMap() {
            MapControl map = new MapControl() { Dock = DockStyle.Fill };
            this.Controls.Add(map);
            map.LegendItemCreating += map_LegendItemCreating;

            VectorItemsLayer fileLayer = new VectorItemsLayer() {
                Colorizer = CreateColorizer(),
                Data = CreateData(filePath),
            };
            map.Layers.Add(fileLayer);

            map.Legends.Add(new ColorScaleLegend() { Header = "GDP in Countries", Layer = fileLayer });
        }

        #region #LegendItemCreating
        void map_LegendItemCreating(object sender, LegendItemCreatingEventArgs e) {
            e.Item.Text = String.Format("{0}$", e.Item.Value);
        }
        #endregion #LegendItemCreating

        private IMapDataAdapter CreateData(string path) {
            Uri baseUri = new Uri(System.Reflection.Assembly.GetEntryAssembly().Location);
            return new ShapefileDataAdapter() { FileUri = new Uri(baseUri, path) };
        }

        private MapColorizer CreateColorizer() {
            
            ChoroplethColorizer colorizer = new ChoroplethColorizer() {
                ApproximateColors = true,
                PredefinedColorSchema = PredefinedColorSchema.Gradient,
                ValueProvider = new ShapeAttributeValueProvider() { AttributeName = "GDP_MD_EST" }
            };
            colorizer.RangeStops.Add(0);
            colorizer.RangeStops.Add(1000);
            colorizer.RangeStops.Add(10000);
            colorizer.RangeStops.Add(100000);
            colorizer.RangeStops.Add(1000000);
            colorizer.RangeStops.Add(10000000);
           
            return colorizer;
        }
    }
}

File: VB/LegendItemCreatingExample/Form1.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.Utils
Imports DevExpress.XtraMap

Namespace LegendItemCreatingExample
    Partial Public Class Form1
        Inherits Form

        Private Const filePath As String = "../../Countries.shp"

        Public Sub New()
            InitializeComponent()

            InitializeMap()
        End Sub

        Private Sub InitializeMap()
            Dim map As New MapControl() With {.Dock = DockStyle.Fill}
            Me.Controls.Add(map)
            AddHandler map.LegendItemCreating, AddressOf map_LegendItemCreating

            Dim fileLayer As New VectorItemsLayer() With {.Colorizer = CreateColorizer(), .Data = CreateData(filePath)}
            map.Layers.Add(fileLayer)

            map.Legends.Add(New ColorScaleLegend() With {.Header = "GDP in Countries", .Layer = fileLayer})
        End Sub

        #Region "#LegendItemCreating"
        Private Sub map_LegendItemCreating(ByVal sender As Object, ByVal e As LegendItemCreatingEventArgs)
            e.Item.Text = String.Format("{0}$", e.Item.Value)
        End Sub
        #End Region ' #LegendItemCreating

        Private Function CreateData(ByVal path As String) As IMapDataAdapter
            Dim baseUri As New Uri(System.Reflection.Assembly.GetEntryAssembly().Location)
            Return New ShapefileDataAdapter() With {.FileUri = New Uri(baseUri, path)}
        End Function

        Private Function CreateColorizer() As MapColorizer

            Dim colorizer As New ChoroplethColorizer() With { _
                .ApproximateColors = True, .PredefinedColorSchema = PredefinedColorSchema.Gradient, .ValueProvider = New ShapeAttributeValueProvider() With {.AttributeName = "GDP_MD_EST"} _
            }
            colorizer.RangeStops.Add(0)
            colorizer.RangeStops.Add(1000)
            colorizer.RangeStops.Add(10000)
            colorizer.RangeStops.Add(100000)
            colorizer.RangeStops.Add(1000000)
            colorizer.RangeStops.Add(10000000)

            Return colorizer
        End Function
    End Class
End Namespace
